tests/clickhouse-test

```diff
@@ -2125,6 +2125,7 @@ class TestCase:
         drop_database_query = f"DROP DATABASE IF EXISTS {database}"
         if args.replicated_database:
             drop_database_query += " ON CLUSTER test_cluster_database_replicated"
+        drop_database_query += " SYNC"
```

Comment:
This is likely to make things much slower in CI, so we might want to only use it before/during `no-parallel` tests. Let's see.